faunadb-scala/src/main/scala/faunadb/FaunaClient.scala

```diff
@@ -37,68 +42,82 @@ object FaunaClient {
     if (metrics ne null) b.withMetrics(metrics)
     if (httpClient ne null) b.withHttpClient(httpClient)
 
-    FaunaClient(b.build)
+    new FaunaClient(b.build)
   }
-
-  def apply(conn: Connection) = new FaunaClient(conn)
 }
 
 /**
- * The Scala native client for FaunaDB.
- *
- * Obtain an instance of the client using the methods on the companion object.
- *
- * The client is asynchronous, so all methods will return a [[scala.concurrent.Future]].
- *
- * Example:
- * {{{
- *  import faunadb.query._
- *
- *  val client = FaunaClient(Connection.builder().withAuthToken("someAuthToken").build))
- *  val response = client.query(Get(Ref("some/ref")))
- * }}}
- */
-class FaunaClient(connection: Connection, json: ObjectMapper = new ObjectMapper) {
+  * The Scala native client for FaunaDB.
+  *
+  * Create a new client using [[FaunaClient.apply]].
+  *
+  * Query requests are made asynchronously: All methods will return a
+  * [[scala.concurrent.Future]].
+  *
+  * Example:
+  * {{{
+  * case class User(ref: Ref, name: String, age: Int)
+  *
+  * val client = FaunaClient(secret = "myKeySecret")
+  *
+  * val fut = client.query(Get(Ref("classes/users/123")))
+  * val instance = Await.result(fut, 5.seconds)
+  *
+  * val userCast =
+  *   for {
+  *     ref <- result("ref").as[Ref]
```

Comment:
Where the `result` comes from? Isn't it `instance` the correct name based on the example above?